ui/pages/confirmations/components/confirm/blockaid-loading-indicator/blockaid-loading-indicator.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+import Preloader from '../../../../../components/ui/icon/preloader';
+import { BlockaidResultType } from '../../../../../../shared/constants/security-provider';
+import { Box } from '../../../../../components/component-library';
+
+import useCurrentConfirmation from '../../../hooks/useCurrentConfirmation';
+import { SecurityAlertResponse } from '../../../types/confirm';
+
+type SignatureSecurityAlertResponsesState = {
+  metamask: {
+    signatureSecurityAlertResponses: Record<string, SecurityAlertResponse>;
+  };
+};
+
+const BlockaidLoadingIndicator = () => {
+  const { currentConfirmation } = useCurrentConfirmation();
+  const securityAlertResponse =
+    currentConfirmation?.securityAlertResponse as SecurityAlertResponse;
+
+  const signatureSecurityAlertResponse = useSelector(
+    (state: SignatureSecurityAlertResponsesState) =>
+      state.metamask.signatureSecurityAlertResponses?.[
+        securityAlertResponse?.securityAlertId as string
+      ],
+  );
+
+  if (
+    signatureSecurityAlertResponse?.result_type === BlockaidResultType.Loading
+  ) {
+    return (
```

Comment:
Minor, but we could also early return `null` to avoid nesting the core logic of the component.

Reply:
PR is updated to address this.